src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/EnvelopeProcessor.java

```diff
@@ -47,12 +49,33 @@ public Envelope parseEnvelope(byte[] metadataStream) throws IOException {
         return EntityParser.parseEnvelopeMetadata(inputStream);
     }
 
-    public Optional<Envelope> hasEnvelopeFailedToUploadBefore(Envelope envelope) {
-        return envelopeRepository.checkLastEnvelopeStatus(
+    /**
+     * Check whether previously attempted envelopes failed.
+     * Only checking last one as it should block any other processing due to exception throw.
+     *
+     * @param envelope details to check against.
+     */
+    public void checkEnvelopeFailedToUploadBefore(Envelope envelope) {
+        List<Envelope> envelopes = envelopeRepository.checkLastEnvelopeByStatus(
             envelope.getContainer(),
             envelope.getZipFileName(),
-            UPLOAD_FAILURE
+            UPLOAD_FAILURE,
+            PageRequest.of(0, 1)
         );
+
+        if (envelopes.size() == 1) {
+            Envelope failedEnvelope = envelopes.get(0);
```

Comment:
Why not check if size() >= 1? I know it should not happen, but better safe than sorry...

Reply:
I'd personally go with `> 0` and leave logging as is.